bugbug/repository.py

```diff
@@ -1543,6 +1544,74 @@ def trigger_pull() -> None:
     trigger_pull()
 
 
+def get_diff(repo_path, original_hash, fix_hash) -> bytes:
+    client = hglib.open(repo_path)
+
+    current_rev = client.identify(id=True)
+
+    try:
+        client.rawcommand([b"shelve"])
+    except hglib.error.CommandError as e:
+        if b"nothing changed" in e.out:
+            logger.info(f"Nothing to shelve: {e}")
+        else:
+            logger.info(f"Error while shelving: {e}")
+            raise
```

Comment:
I don't think logging is the right action here. You might want to consider using raise-from.

Reply:
Done in 66108ad83f3e2579af14ee983758cc3b5ff7fcf9